lib/repl.js

```diff
@@ -901,8 +901,10 @@ const simpleExpressionRE =
     /(?:[a-zA-Z_$](?:\w|\$)*\.)*[a-zA-Z_$](?:\w|\$)*\.?$/;
 
 function intFilter(item) {
-  // filters out anything not starting with A-Z, a-z, $ or _
-  return /^[A-Za-z_$]/.test(item);
+  // Filters the elements that pass the regexp:
+  // 1. Starting with A-Z, a-z, $ or _
+  // 2. Following with A-Z, a-z, 0-9, $ or _
+  return /^[A-Za-z_$][A-Za-z0-9_$]*$/.test(item);
```

Comment:
With acorn in tree, we could actually do something like

https://github.com/devsnek/node-repl-prototype/blob/5e72bed5a127c0b3e1c7289d75c3eb157db43522/src/util.js#L3-L24

for higher accuracy.